Assets/MRTK/SDK/Editor/Inspectors/Experimental/Elastics/ElasticsManagerInspector.cs

```diff
@@ -0,0 +1,143 @@
+            elasticTypes.intValue = (int)(TransformFlags)newElasticTypesValue;
+
+            // If the particular elastic type is requested, we offer the user the ability
+            // to configure the elastic system.
+            TransformFlags currentFlags = (TransformFlags)elasticTypes.intValue;
+
+            translationElasticFoldout = DrawElasticConfiguration<ElasticConfiguration>(
+                "Translation Elastic",
+                translationElasticFoldout,
+                translationElasticConfigurationObject,
+                translationElasticExtent,
+                TransformFlags.Move,
+                currentFlags);
+
+            rotationElasticFoldout = DrawElasticConfiguration<ElasticConfiguration>(
+                "Rotation Elastic",
+                rotationElasticFoldout,
+                rotationElasticConfigurationObject,
+                rotationElasticExtent,
+                TransformFlags.Rotate,
+                currentFlags);
+
+            scaleElasticFoldout = DrawElasticConfiguration<ElasticConfiguration>(
+                "Scale Elastic",
+                scaleElasticFoldout,
+                scaleElasticConfigurationObject,
+                scaleElasticExtent,
+                TransformFlags.Scale,
+                currentFlags);
+
+            serializedObject.ApplyModifiedProperties();
+        }
+
+        private bool DrawElasticConfiguration<T>(
+            string name,
+            bool expanded,
+            SerializedProperty elasticProperty,
+            SerializedProperty extentProperty,
+            TransformFlags requiredFlag,
+            TransformFlags providedFlags) where T : ElasticConfiguration
+        {
+            if (providedFlags.HasFlag(requiredFlag))
+            {
+                bool result = false;
+                using (new EditorGUI.IndentLevelScope())
+                {
+                    result = InspectorUIUtility.DrawScriptableFoldout<T>(
+                        elasticProperty,
+                        name,
+                        expanded);
+                    EditorGUILayout.PropertyField(extentProperty, includeChildren: true);
+                }
+                return result;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+
+        /// <summary>
+        /// Draws a property of type ElasticsManager, showing a button to create a component to link to
+        /// </summary>
+        /// <param name="elasticsManager">Elasticsmanager property of component this ui field gets drawn in.</param>
+        /// <param name="hostObject">GameObject the component this is added to is attached to.</param>
+        public static void DrawElasticsManagerLink(SerializedProperty elasticsManager, GameObject hostObject)
+        {
+            EditorGUILayout.LabelField(new GUIContent("Elastics"), EditorStyles.boldLabel);
+            {
```

Comment:
Is this extra scope mostly there as a visual aid (i.e. "everything within here shows up in some elastics context?")

Reply:
Yes. I usually put everything that belongs to one section /heading in one scope so it's easier to read in long inspector files

Reply:
it's probably not needed in that method? - should i remove it in the next PR?